web/packages/teleterm/src/main.ts

```diff
@@ -356,4 +360,8 @@ function launchDeepLink(logger: Logger, rawUrl: string): void {
 
     logger.error(`Skipping deep link launch, ${reason}`);
   }
+
+  // Always pass the result to the frontend app so that the error can be shown to the user.
```

Comment:
Yes, exactly.

This is a typical Electron app architecture where the main process ([`main.ts`](https://github.com/gravitational/teleport/blob/db4fe9ebbb56893dce75c51248e592810e47ae75/web/packages/teleterm/src/main.ts)) opens a new browser window (a new renderer process) which runs some privileged preload code that has access to Node.js APIs ([`preload.ts`](https://github.com/gravitational/teleport/blob/db4fe9ebbb56893dce75c51248e592810e47ae75/web/packages/teleterm/src/preload.ts)) as well as the actual frontend app ([`ui/boot.tsx`](https://github.com/gravitational/teleport/blob/db4fe9ebbb56893dce75c51248e592810e47ae75/web/packages/teleterm/src/ui/boot.tsx)). The code of the frontend app doesn't differ that much from a regular app, but [the preload code can expose some Node.js APIs](https://github.com/gravitational/teleport/blob/db4fe9ebbb56893dce75c51248e592810e47ae75/web/packages/teleterm/src/preload.ts#L49-L54) to the frontend app [as properties defined on `window`](https://github.com/gravitational/teleport/blob/db4fe9ebbb56893dce75c51248e592810e47ae75/web/packages/teleterm/src/ui/boot.tsx#L57).

For example, the gRPC client we use in Connect is actually a Node.js gRPC client running as a part of the frontend app. There are several limitations when it comes to interactions between the preload code and the frontend app code (because [they run in different contexts](https://www.electronjs.org/docs/latest/api/context-bridge)) and between the main process and the renderer process (because [they're separate OS processes](https://www.electronjs.org/docs/latest/tutorial/ipc)).